oeps/oep-0005.rst

```diff
@@ -0,0 +1,182 @@
+=======================================
+
++---------------+-------------------------------------------+
+| OEP           | :doc:`OEP-5 </oeps/oep-0005>`             |
++---------------+-------------------------------------------+
+| Title         | Pre-built Developer Environments          |
++---------------+-------------------------------------------+
+| Last Modified | 2016-06-28                                |
++---------------+-------------------------------------------+
+| Author        | Calen Pennington <cale@edx.org>           |
++---------------+-------------------------------------------+
+| Arbiter       |                                           |
++---------------+-------------------------------------------+
+| Status        | Draft                                     |
++---------------+-------------------------------------------+
+| Type          | Best Practice                             |
++---------------+-------------------------------------------+
+|  Created      | 2016-06-28                                |
++---------------+-------------------------------------------+
+
+Abstract
+========
+
+Open edX is a large ecosystem of smaller services. In order to facilitate
+newcomers working in that system, each service will provide a prebuilt environment
+that is packaged, published, and ready to be used for development.
+
+Motivation
+==========
+
+Open edX and the edx-platform repository have gone through several iterations
+of developer setup instructions.
+
+The first set of instructions detailed how to install edx-platform and all of
+its dependencies on your local workstation. As the number of developers and
+developer environments increased, so did the workload of maintaining instructions
+to work around all of the ideosyncacies in those environments. These instructions
+weren't useful for production installations, which were standardized on a
+particular release of the Ubuntu operating system.
+
+The second, current, method of development is referred to as Devstack. This is
+a `Vagrant`_ image that comes prepackaged with installations of edx-platform,
+forums, and an increasingly large number of other services, as well as single
+shared instances of the databases that they all need to run. This marked a significant
+improvement over the ad-hoc installation methods, because it was able to use
+the infrastructure that was built out for installing Open edX in production.
+However, there are several problems with the current monolithic `Vagrant`_ image
+which we would like to address:
+
+1. As new services are added to the Open edX ecosystem, the `Vagrant`_ image
+   grows in size, requiring more system memory from the host machine.
+2. `Vagrant`_ imposes noticable overhead during filesystem access, which especially
+   hurts any development that it doing file-heavy operations (such as manipulating
+   static assets).
+3. The monolithic `Vagrant`_ image isn't particularly representative of production
```

Comment:
I think this is a bit of an overstatement. My understanding is that it is representative of production based on system requirements, but notsomuch on system footprint.